core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/ClusterStartHandler.java

```diff
@@ -131,21 +136,37 @@ public void accept(Event<ClusterStartRequest> event) {
             } else {
                 requestId = apiConnectors.getConnector(stack).startCluster();
             }
-            if (stackUtil.stopStartScalingEntitlementEnabled(stack)) {
-                List<String> decommissionedHostsFromCM = apiConnectors.getConnector(stack).clusterStatusService().getDecommissionedHostsFromCM();
-                // TODO: CB-15341 Is this a sufficient check for compute hostgroup in the long run?
-                List<String> decommissionedComputeHosts = decommissionedHostsFromCM.stream().filter(h -> h.contains("compute")).collect(Collectors.toList());
-                if (!decommissionedComputeHosts.isEmpty()) {
-                    apiConnectors.getConnector(stack).clusterCommissionService().recommissionHosts(decommissionedComputeHosts);
-                }
-            }
+            handleStopStartScalingFeature(stack, blueprintProcessor);
             result = new ClusterStartResult(request, requestId);
         } catch (Exception e) {
             result = new ClusterStartResult(e.getMessage(), e, request);
         }
         eventBus.notify(result.selector(), new Event<>(event.getHeaders(), result));
     }
 
+    @VisibleForTesting
+    void handleStopStartScalingFeature(Stack stack, CmTemplateProcessor blueprintProcessor) {
+        if (stackUtil.stopStartScalingEntitlementEnabled(stack)) {
+            List<String> decommissionedHostsFromCM = apiConnectors.getConnector(stack).clusterStatusService().getDecommissionedHostsFromCM();
+            Set<String> computeGroups = getComputeHostGroups(blueprintProcessor);
+            if (computeGroups.isEmpty() || decommissionedHostsFromCM.isEmpty()) {
```

Comment:
Nit: Separating this check can avoid a call to CM if there are not compute groups.